test/parallel/test-http2-compat-serverresponse-finished.js

```diff
@@ -9,36 +9,36 @@ const net = require('net');
 
 // Http2ServerResponse.finished
 const server = h2.createServer();
-server.listen(0, common.mustCall(function() {
+server.listen(0, common.mustCall(() => {
   const port = server.address().port;
-  server.once('request', common.mustCall(function(request, response) {
+  server.once('request', common.mustCall((request, response) => {
     assert.ok(response.socket instanceof net.Socket);
     assert.ok(response.connection instanceof net.Socket);
     assert.strictEqual(response.socket, response.connection);
 
-    response.on('finish', common.mustCall(function() {
-      assert.strictEqual(response.socket, undefined);
-      assert.strictEqual(response.connection, undefined);
+    response.on('finish', common.mustCall(() => {
+      assert.ok(!response.socket);
```

Comment:
I see this kind of change being done frequently recently and I think this is not always a good idea. If there are other tests that cover the real value then it's ok, if not we are hiding the real type coercing to a boolean.

What I mean is that if a regression is introduced where `response.socket` is set to `null` (or any other falsy value) instead of `undefined`, with this change, we will not notice.

Reply:
@lpinca I got your point. I even thought about it before I did the changes, but in this context, when the response is `finished`, its gonna be `undefined`. So `socket` and `connection` are no more valid, I felt it was fine to use `ok`. But yeah, if some other logic expects the `socket` to be `undefined` and we pass `null`, we gonna cause a regression. Good point.